spec/lib/http/response/caching_spec.rb

```diff
@@ -0,0 +1,202 @@
+
+    it "is not cacheable" do
+      expect(subject.cacheable?).to be_falsy
+    end
+
+    it "is not stale" do
+      expect(subject.stale?).to be_falsy
+    end
+
+    it "is not expired" do
+      expect(subject.expired?).to be_falsy
+    end
+
+    it "is expected to be 0 seconds old" do
+      expect(subject.current_age).to be_within(1).of(0)
+    end
+  end
+
+  describe "basic 200 response w/ max age" do
+    let(:cache_control) { "max-age=100" }
+
+    it "is not cacheable" do
+      expect(subject.cacheable?).to be_truthy
+    end
+
+    it "is not stale" do
+      expect(subject.stale?).to be_falsy
+    end
+
+    it "is not expired" do
+      expect(subject.expired?).to be_falsy
+    end
+
+    it "is expected to be 0 seconds old" do
+      expect(subject.current_age).to be_within(1).of(0)
+    end
+  end
+
+  describe "basic 200 response w/ public & max age" do
+    let(:cache_control) { "public, max-age=100" }
+
+    it "is not cacheable" do
+      expect(subject.cacheable?).to be_truthy
+    end
+
+    it "is not stale" do
+      expect(subject.stale?).to be_falsy
+    end
+
+    it "is not expired" do
+      expect(subject.expired?).to be_falsy
+    end
+
+    it "is expected to be 0 seconds old" do
+      expect(subject.current_age).to be_within(1).of(0)
+    end
+
+    context "with age of max-age + 1 seconds" do
+      let(:headers) { {"cache-control" => cache_control, "age" => "101"} }
+
+      it "is stale" do
+        expect(subject.stale?).to be_truthy
+      end
+
+      it "is expired" do
+        expect(subject.expired?).to be_truthy
+      end
+
+      it "is expected to be max-age + 1 seconds old" do
+        expect(subject.current_age).to be_within(1).of(101)
+      end
+    end
+
+    context "after max-age + 1 seconds" do
+      before do
+        subject.received_at = subject.requested_at = (Time.now - 101)
+      end
+
+      it "is stale" do
+        expect(subject.stale?).to be_truthy
+      end
+
+      it "is expired" do
+        expect(subject.expired?).to be_truthy
+      end
+
+      it "is expected to be max-age + 1 seconds old" do
+        expect(subject.current_age).to be_within(1).of(101)
+      end
+    end
+  end
+
+  describe "basic 400 response " do
+    let(:response) { HTTP::Response.new(400, "http/1.1", {}, "") }
+
+    it "is not cacheable" do
+      expect(subject.cacheable?).to be_falsy
+    end
+
+    it "is not stale" do
+      expect(subject.stale?).to be_falsy
+    end
+
+    it "is not expired" do
+      expect(subject.expired?).to be_falsy
+    end
+
+    it "is expected to be 0 seconds old" do
+      expect(subject.current_age).to be_within(1).of(0)
+    end
+  end
+
+  # Background
+  let(:cache_control) { "" }
+  let(:headers) { {"cache-control" => cache_control} }
+  let(:response) { HTTP::Response.new(200, "http/1.1", headers, "") }
+
+  describe "#caching" do
+    subject(:caching_response) { response.caching }
+    it { is_expected.to be caching_response }
+  end
```

Comment:
i think it is a bit more like `it "returns itself"`. I changed it though, because that is an implementation detail not an essential part of the behavior.